calcApp.js

```diff
@@ -0,0 +1,435 @@
+'use strict';
+// This application has been designed within pattern "Command"
+
+class CalcApp {
+    constructor(container) {
+        // this object contains interface methods of realization
+        let calculator = new Calculator();
+
+        // User interface.
+        // Making this property as a public in case of testing aims
+        this.calculatorUI = new CalculatorUI(calculator, container);
+    }
+}
+
+class CalculatorUI {
+    constructor(calculator, calcContainer) {
+        this._calculator = calculator;
+        this._calcContainer = calcContainer;
+        this._screen = this._calcContainer.querySelector('.calculator__screen');
+        this._screen.value = 0; // default value
+
+        // helper flags for calculator buttons
+        this._firstOperandInputted = false;
+        this._equallyInputted = false;
+        this._currentOperator = null; // the current operator, which is going to be executed
+        this._inputUpdated = false; // it changes, when user inputs new character
+
+
+        // add event handlers for mobile devices
+        if (/Android|webOS|iPhone|iPad|iPod|BlackBerry|IEMobile|Opera Mini/i.test(navigator.userAgent)) {
+
+            this._calcContainer.addEventListener('mousedown', () => this._processMouseEvent(event));
+
+        // for desktop
+        } else {
+            this._screen.focus(); // default
+            this._calcContainer.addEventListener('click', () => this._processMouseEvent(event));
+            this._calcContainer.addEventListener('click', () => this._screen.focus());
+            this._screen.addEventListener('keypress', () => this._processKeyboardEvent(event));
```

Comment:
I would rather use `this._processKeyboardEvent.bind(this)`
It works in all browsers